controllers/users.go

```diff
@@ -0,0 +1,90 @@
+package controllers
+
+import (
+	"costmanagement/db"
+	"costmanagement/models"
+	"costmanagement/models/dtos"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+//POST /users
+// create new user
+func CreateUser(c *gin.Context) {
+	//validate input
+	var input dtos.CreateUserInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
+		return
+	}
+
+	//create user
+	user := models.User{
+		Name:     input.Name,
+		Email:    input.Email,
+		Phone:    input.Phone,
+		Password: input.Password,
+	}
+	db.DB.Create(&user)
+
+	c.JSON(http.StatusOK, gin.H{"data": user})
+}
+
+//PATCH /users/:id
+//Update  a user
+func UpdateUser(c *gin.Context) {
+	// Validate input
+	var input dtos.UpdateUserInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	// Get model if exist
+	var user models.User
+	if err := db.DB.Where("id = ?", c.Param("id")).First(&user).Error; err != nil {
```

Comment:
you are accessing `db.DB.Where()`, how your system or code recognize that you are searching in User table. Suppose we have multiple table exist including users table, then how your query recognize on which table it should query?

Reply:
this query is taking the first "user" model .
.First(&user)

and user is a variable of User type

Reply:
### userDbContext ###
will contain the user table ??

Reply:
yes. like `userDbContext := db.Table(domains.UserTable)`. Is it more readable for anyone who will letter handle this project or want to contribute in this project?

Reply:
what if store all users like GET operation and then run query on the list?

Reply:
don't get you

Reply:
```func FindUsers(c *gin.Context) {
	var users []models.User
	db.DB.Find(&users)
}
```
 i want to run query in
`` users ```

Reply:
I already agreed with your concern. I am just talking about code readability. When someone read this code, it is very much possible to get confused.

Reply:
Should I merge?

Reply:
i am confused about 
```
domains.UserTable
```

Reply:
oh, this is just a string, where UserTable = "users"

Reply:
name of that table, defined in a constant